### ISSUE TITLE
Azure DevOps HTTPS and SSH git urls not being matched

### ISSUE THREAD

**Issue by edtownend** [getsentry/sentry-cli#550]
Hi, I'm trying to automatically associate commits with an Azure DevOps repository. The trouble seems to be that sentry is only aware of the HTTPS endpoint for the repository, and my local git is using SSH.

As you see the repo is added in sentry: (I've masked the repo name)
```
sentry-cli repos list
+------+--------------+-------------------------------------------+
| Name | Provider     | URL                                       |
+------+--------------+-------------------------------------------+
| XXX  | Azure DevOps | https://xxx.visualstudio.com/XXX/_git/XXX |
+------+--------------+-------------------------------------------+
```

But the command `sentry-cli releases set-commits --log-level=DEBUG --auto ${version}` results in

```
INFO    2019-06-11 11:20:23.006086 +01:00 Resolving HEAD (XXX@HEAD)
DEBUG   2019-06-11 11:20:23.006334 +01:00   Got reference URL for repo XXX: https://XXX.visualstudio.com/XXX/_git/XXX
DEBUG   2019-06-11 11:20:23.006347 +01:00   Looking for reference URL https://XXX.visualstudio.com/XXX/_git/XXX
DEBUG   2019-06-11 11:20:23.006636 +01:00   not a match: ssh://XXX@vs-ssh.visualstudio.com:22/_ssh/XXX != https://XXX.visualstudio.com/XXX/_git/XXX
INFO    2019-06-11 11:20:23.014101 +01:00   -> no matching revision found
DEBUG   2019-06-11 11:20:23.014862 +01:00 error: running update nagger
error: Could not determine any commits to be associated automatically.
```

Basically, it looks like because DevOps' HTTPS urls include the company name in the subdomain (`https://XXX.visualstudio.com/XXX/_git/XXX`) and the SSH ones don't (`ssh://XXX@vs-ssh.visualstudio.com:22/_ssh/XXX`), the repo isn't getting matched.

Thanks! Please let me know if I can offer any more info.

**Comment by markstory** [comment]
Are all the XXX containing the same value in your actual data?

**Comment by edtownend** [comment]
Sorry, no, there are two different values - the instances of XXX below are all equal as are YYY.

```
ssh://XXX@vs-ssh.visualstudio.com:22/_ssh/YYY
https://XXX.visualstudio.com/YYY/_git/YYY
```

**Comment by markstory** [comment]
How does one go about getting that kind of SSH url from Azure?  In my azure account I only see `git@ssh.dev.azure.com:v3/<project>/<repo>` style URLs.

**Comment by edtownend** [comment]
![Screenshot 2019-06-17 at 13 48 56](https://user-images.githubusercontent.com/2234457/59605555-f7848800-9106-11e9-8a6e-d42aa15155aa.png)

This is what it offers for me in the clone dropdown.

**Comment by edtownend** [comment]
![Screenshot 2019-06-17 at 13 48 56](https://user-images.githubusercontent.com/2234457/59605730-4b8f6c80-9107-11e9-889e-e3e7910f455b.png)

This is what it shows for me in the clone dropdown.

**Comment by edtownend** [comment]
![Screenshot 2019-06-17 at 13 48 56](https://user-images.githubusercontent.com/2234457/59605794-7679c080-9107-11e9-9173-3286c1b11036.png)

Third time lucky at blanking the sensitive bits! This is what it shows for me in the clone dropdown.

**Comment by markstory** [comment]
Interesting. That same point in the UI gives me `git@ssh.dev.azure.com` urls. Perhaps the URL style you have comes from private projects or on a paid account. I'll see if I can get our URL normalizer to handle this new style of URL from azure.

**Comment by edtownend** [comment]
I've made matters way worse, I found a button in DevOps' organization settings to update the URLs to the new style (like yours), then updated my local git's remote url to reflect it. The trouble is, even since disconnecting and reconnecting the Sentry/Azure DevOps integration, Sentry is still using the old style URL to compare.

Now the output from `sentry-cli repos list` is the same as at the start, and the set-commits command gives the following:

`not a match: git@ssh.dev.azure.com:v3/<project>/<repo>/<repo> != https://<project>.visualstudio.com/<repo>/_git/<repo>`

In summary - Is there any way to force Sentry's DevOps integration to 'see' that the URLs have updated?

Thanks

**Comment by markstory** [comment]
There isn't a way to update a repository's URL outside of removing it and re-adding it. This will also delete existing commit information though :cry:

**Comment by edtownend** [comment]
That's not a problem, there was no commit info to lose in Sentry. It didn't work the first time but has done after making a cuppa. Now Sentry is showing the new url, but still won't match it to the local git remote!

`not a match: git@ssh.dev.azure.com:v3/<project>/<repo>/<repo> != https://dev.azure.com/<project>/<repo>/_git/<repo>`

**Comment by markstory** [comment]
Currently sentry-cli doesn't have any special casing for dev.azure.com URLs, however, I can add that as they will be quite common for Azure users.

**Comment by edtownend** [comment]
I've updated to 1.47.0 and I'm afraid the repos are still not getting matched.

```
  DEBUG   2019-07-24 14:10:39.141992 +01:00   Got reference URL for repo reponame: https://dev.azure.com/companyname/reponame/_git/reponame
  DEBUG   2019-07-24 14:10:39.142004 +01:00   Looking for reference URL https://dev.azure.com/companyname/reponame/_git/reponame
  DEBUG   2019-07-24 14:10:39.142203 +01:00   not a match: git@ssh.dev.azure.com:v3/companyname/reponame/reponame != https://dev.azure.com/companyname/reponame/_git/reponame
```

**Comment by markstory** [comment]
@edtownend Interesting. That is basically the same as an existing test https://github.com/getsentry/sentry-cli/blob/5a6e033db43099c86026a425be9cfe767c057306/src/utils/vcs.rs#L512-L515

Did I get something wrong with that test?

**Comment by rodrigomocca** [comment]
Hi,
I'm dealing with this issue right now.
```
  INFO    2019-07-24 15:02:01.805066400 -03:00 Resolving HEAD (tracking_module@HEAD)
  DEBUG   2019-07-24 15:02:01.807080700 -03:00   Got reference URL for repo tracking_module: https://qualdel.visualstudio.com/QualDel%20Projects/_git/tracking_module
  DEBUG   2019-07-24 15:02:01.808064500 -03:00   Looking for reference URL https://qualdel.visualstudio.com/QualDel%20Projects/_git/tracking_module
  DEBUG   2019-07-24 15:02:01.810063400 -03:00   not a match: https://qualdel@dev.azure.com/qualdel/QualDel%20Projects/_git/tracking_module != https://qualdel.visualstudio.com/QualDel%20Projects/_git/tracking_module
```
Looking at my organization settings I have this:
![image](https://user-images.githubusercontent.com/4087916/61817288-926a3380-ae24-11e9-8e98-2f63e2a63d05.png)

How can I set the URL to my repo with dev.azure.com format instead of visualstudio.com?

**Comment by edtownend** [comment]
@rodrigomocca There should be a button somewhere in Project Settings to enable the new URL format - [this guide](http://www.codewrecks.com/blog/index.php/2018/11/20/set-new-azure-devops-url-for-your-account/) seems to cover it.

@markstory I can't see anything wrong with that test - the only thing I can think is my repo name contains a space and capital letters so looks like Repo%20Name but does match on each side of the comparison.

**Comment by rodrigomocca** [comment]
@edtownend Thanks, but I have this enabled in my azure devops account. But sentry still uses the old url, so I think this could be the problem. There is any way to tell sentry to use dev.azure.com instead visualstudio.com? Or there is any way to sentry-cli match this 2 urls as equal?

**Comment by edtownend** [comment]
@rodrigomocca Mark answered that above - remove the repository from Sentry and re-add, but it does lose any already added commit history. It took a couple of goes to get it to work for me, or maybe I hadn't waited long enough.

**Comment by markstory** [comment]
@edtownend A urlencoded space could cause a mismatch. Do both the local clone url and API response from sentry have the encoded space?

**Comment by edtownend** [comment]
@markstory yes, both look identical with the caps and %20. Happy to email you the full unredacted logs if it helps?

**Comment by markstory** [comment]
@edtownend That would be helpful. mark[at]sentry.io is my email address.
